spec/Classes.tex

```diff
@@ -784,12 +787,54 @@ \subsection{The Method Receiver and the {\em this} Argument}
 is \chpl{c1}.
 \end{chapelexample}
 
-The optional \sntx{this-intent} declares the intent by which the receiver
-argument will be passed.  When the value of \sntx{this-intent} is \chpl{param},
-it implies that the function can only be applied to param objects of the given
-type binding.  If the value is \chpl{ref}, the argument will be passed by
-reference---allowing modification of \chpl{this}.  Otherwise, the argument will
-be passed with the blank intent as specified in \rsec{The_Blank_Intent}.
+The optional \sntx{this-intent} declares how the receiver argument is
+constrained or will be passed for a given method call.
+
```

Comment:
Almost certainly not up to Lydia's standards due to time constraints, but I've added a test that shows an error when invoking a "static" method on an instance rather than a type.